homeassistant/components/ialarm/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""iAlarm integration."""
+import logging
+
+from async_timeout import timeout
+from pyialarm import IAlarm
+
+from homeassistant.components.alarm_control_panel import SCAN_INTERVAL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PORT,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DATA_COORDINATOR, DOMAIN
+
+PLATFORM = "alarm_control_panel"
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _get_device_mac(hass: HomeAssistant, ialarm) -> str:
+    try:
+        async with timeout(10):
+            return await hass.async_add_executor_job(ialarm.get_mac)
```

Comment:
Verified this it obtained via the api 👍

Reply:
We shouldn't raise `UpdateFailed` here as thats only for the coordinator.

Reply:
Ah okay, fixed it with the latest revision.